src/main/scala/beam/utils/ScenarioReaderCsv.scala

```diff
@@ -0,0 +1,378 @@
+    val householdPersons = ScenarioReaderCsv.readPersonsFile(
+      personFilePath,
+      scenario.getPopulation,
+      BeamMode.allBeamModes.map(_.value).mkString(",")
+    )
+
+    logger.info("Reading plans...")
+    ScenarioReaderCsv.readPlansFile(planFilePath, scenario.getPopulation, beamServices)
+
+    logger.info("Total persons loaded {}", scenario.getPopulation.getPersons.size())
+    logger.info("Checking persons without selected plan...")
+
+    val listOfPersonsWithoutPlan: ListBuffer[Id[Person]] = ListBuffer()
+    scenario.getPopulation.getPersons.forEach {
+      case (pk: Id[Person], pv: Person) if (pv.getSelectedPlan == null) => listOfPersonsWithoutPlan += pk
+      case _                                                            =>
+    }
+
+    logger.info("Removing persons without plan {}", listOfPersonsWithoutPlan.size)
+    listOfPersonsWithoutPlan.foreach { p =>
+      val _hId = scenario.getPopulation.getPersonAttributes.getAttribute(p.toString, "householdId")
+
+      scenario.getPopulation.getPersonAttributes.removeAllAttributes(p.toString)
+
+      scenario.getPopulation.removePerson(p)
+
+      val hId = Id.create(_hId.toString, classOf[Household])
+      householdPersons.get(hId) match {
+        case Some(persons: ListBuffer[Id[Person]]) =>
+          persons -= p
+        case None =>
+      }
+    }
+
+    logger.info("Reading Households...")
+    ScenarioReaderCsv.readHouseholdsFile(
+      householdFilePath,
+      scenario,
+      beamServices,
+      householdPersons.par,
+      units.par,
+      buildings.par,
+      parcelAttrs.par
+    )
+
+    logger.info("Total households loaded {}", scenario.getHouseholds.getHouseholds.size())
+    logger.info("Checking households without members...")
+    val listOfHouseholdsWithoutMembers: ListBuffer[Household] = ListBuffer()
+    scenario.getHouseholds.getHouseholds.forEach {
+      case (hId: Id[Household], h: Household) if (h.getMemberIds.size() == 0) => listOfHouseholdsWithoutMembers += h
+      case _                                                                  =>
+    }
+
+    logger.info("Removing households without members {}", listOfHouseholdsWithoutMembers.size)
+    listOfHouseholdsWithoutMembers.foreach { h =>
+      removeHouseholdVehicles(h)
+      scenario.getHouseholds.getHouseholdAttributes.removeAllAttributes(h.getId.toString)
+      scenario.getHouseholds.getHouseholds.remove(h.getId)
+    }
+
+    logger.info("The scenario loading is completed..")
+  }
+
+  def removeHouseholdVehicles(household: Household) = {
+    household.getVehicleIds.forEach(
+      vehicleId => beamServices.privateVehicles.remove(Id.create(vehicleId.toString, classOf[BeamVehicle]))
+    )
+  }
+}
+
+object ScenarioReaderCsv {
+  private val logger = LoggerFactory.getLogger(this.getClass)
+
+  var vehicleCounter = 1;
+
+  def readPersonsFile(
+    filePath: String,
+    population: Population,
+    modes: String
+  ): TrieMap[Id[Household], ListBuffer[Id[Person]]] = {
```

Comment:
Is there any particular reason to use `TrieMap` from `scala.collection.concurrent.TrieMap`? I don't see any concurrent read/write there.